.github/actions/setup-dfx/action.yml

```
@@ -0,0 +1,32 @@
+name: 'Setup DFX'
+description: Setup DFX
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/cache@v2
+      with:
+        path: |
+          ~/dfx
+          ~/.cache/dfinity
+        key: dfx-cache-0.11.1-2
+    # This step hangs on Github actions on Darwin for some reason, that
+    # is why we run this only on Linux for now
+    - name: Install DFX
+      shell: bash
+      env:
+        DFX_VERSION: 0.11.1
+      run: |
+        if [ -f ~/dfx/dfx ]
+        then
+          echo "DFX restored from cache"
+          ~/dfx/dfx --version
+        else
+          # sent env variable here so that '~' gets expanded
```

```suggestion
          # set env variable here so that '~' gets expanded
```

here as opposed to where? `~` should get expanded everywhere (as long as you don't wrap in it quotes or double quotes)